src/Integration.UnitTests/MefServices/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -183,14 +184,16 @@ public void ActiveSolutionBoundTracker_Changes()
 
             using (var testSubject = new ActiveSolutionBoundTracker(this.host, this.activeSolutionTracker, loggerMock.Object))
             {
-                var solutionBindingChangedEventCount = 0;
-                testSubject.SolutionBindingChanged += (obj, args) => { solutionBindingChangedEventCount++; };
+                var eventCounter = new EventCounter(testSubject);
 
                 // Sanity
                 testSubject.CurrentConfiguration.Mode.Should().Be(SonarLintMode.LegacyConnected, "Initially bound");
                 this.errorListController.RefreshCalledCount.Should().Be(0);
                 this.errorListController.ResetCalledCount.Should().Be(0);
-                solutionBindingChangedEventCount.Should().Be(0, "no events raised during construction");
+                eventCounter.PreSolutionBindingChangedCount.Should().Be(0, "no events raised during construction");
+                eventCounter.SolutionBindingChangedCount.Should().Be(0, "no events raised during construction");
+                eventCounter.PreSolutionBindingUpdatedCount.Should().Be(0, "no events raised during construction");
+                eventCounter.SolutionBindingUpdatedCount.Should().Be(0, "no events raised during construction");
```

Comment:
I've updated all of the existing tests to use the new helper class, and to check all of the events.